src/components/Form/Form.cy.tsx

```diff
@@ -42,23 +42,22 @@ describe('Form', () => {
     });
 
     it('should submit the form', () => {
-        const onSubmitStub = cy.stub().as('onSubmitStub');
-
         cy.mount(
-            <Form
-                onSubmit={(event) => {
-                    event.preventDefault();
-                    onSubmitStub();
-                }}
-            >
+            <Form method="POST" action="/form">
                 <TextInput id="label" value="some value" />
                 <Button type={ButtonType.Submit}>Submit</Button>
             </Form>,
         );
 
         cy.get(BUTTON_ID).should('exist');
         cy.get(FORM_ID).should('exist');
-        cy.get(BUTTON_ID).click();
-        cy.wrap(onSubmitStub).should('have.been.calledOnce');
+        cy.intercept('POST', '/form', {
```

Comment:
Network calls have more chances of flakeyness, what about using `onSubmit` to test it?

Reply:
Maybe, I followed the docs and it seemed to be the best way to verify the data is actually been submitted. But I can improve the tests later when we have more inputs and the FormField build. For now, I would leave it that way because it's side work and delaying my OKR focus. I'll create a task though.